am.test_qm_metadata

amikulichmines opened the pull request:
Created class to read in Excel files, and a test to
make sure it works correctly.

amikulichmines committed:
Created test and the beginnings of a class to read in and merge multiple excel files.

amikulichmines committed:
Added write and merge_conflict functions

amikulichmines committed:
Added write and merge_conflict functions

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+import pandas as pd
+
```
Why is this in the tests directory? If this is part of the code, it should be in the src/confluence subtree.

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+
+
+class Excel_reader():
+    def __init__(self, *filenames):
```
This breaks atomicity. A reader should read one file and produce one or more objects. You would then use a function or object to merge/join the results from multiple reads.

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        self.combined = None
+
+    def _open_excel(self, filename):
+        self.ifs = pd.read_excel(filename)
```
The functionality in pandas read_excel comes from the options in this function. We will rarely have an excel file were the default options are sufficient. Replace this with

```
def read_excel(self, filename, **kwds):
    self._df = pd.read_excel(filename, **kwds)
    return self._df
```

Note the change in the name. Since this is calling the pandas API, this is just a wrapper/convenience function around this pandas function.

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+import pandas as pd
+
+
+class Excel_reader():
```
Is this a base level object? Isn’t this a type of reader, and as such, shouldn’t this derive from, at least, BaseReader?

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        self.ifs = pd.read_excel(filename)
+        return self.ifs
+
+    def _close(self):
```
`pd.read_excel` does not return an open file handle, so there is nothing to close.

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        :param column: Name of the column being checked
+
+        :return: none
+        """
```
Shouldn’t filename and dataframe have been defined from an early call to `read`? We should only need the column name. When do we need to know when any entry in a column is missing, but we don’t care which one?

This might better be named `any_empty`.

You can simplify this,

```
return self.df[column].isna().any()

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        :param filename: name of the file
+        :param dataframe: excel spreadsheet data
+        :return: none
+        """
```
We shouldn’t need filename or dataframe. This, too, should return a Boolean: True if duplicates are found, False otherwise, e.g.

```
return self.df.duplicated().any()
```

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+                :param dataframe: excel spreadsheet data
+                :return: dataframe with no duplicates
+                """
+
```
This doesn’t update the existing dataframe. This should probably wrap the functionality of `drop_duplicates`

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        :param dataframe: excel spreadsheet data
+        :return: none
+        """
+
```
How is this different from `_check_for_duplicates`?

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        function: write a dataframe to excel
+        :param filename:
+        :return: none
+        """
```
Why would we write this to “Sheet1” by default?

bkappes commented on tests/ExcelReader.py:
```diff
@@ -0,0 +1,102 @@
+        """
+        creates a merged dataframe. As of July 14, it has not yet been coded to delete duplicate rows.
+        :return: merged dataframe
+        """
```
Have you looked into pandas join operations?

amikulichmines committed:
Fixed some of the tests to allow it to identify which two files were causing a merge
conflict.

amikulichmines committed:
Mostly finished the merge functions.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the excel_parser function, data from identically named worksheets are grouped together and analyzed for merge conflicts. In this code, if a conflict is detected, it prompts user input to determine the right value.

As for the argparse functions, it has the -o, --output function which allows the user to specify the output file. It also has the -i, --input function to specify the type of the file ('xlsx', 'txt', etc.) In the future, it will also have a -m, --merge-default function to specify a default resolution to a conflict, as well as a --output-format function to specify the type of the output file.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

The read_text and write_text functions are set up as well, and have been torture tested to some degree. One issue I noticed was that it did read in all the numbers as strings, so if this is a problem moving forward, I can augment it to read them in as floats.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

The read_text and write_text functions are set up as well, and have been torture tested to some degree. One issue I noticed was that it did read in all the numbers as strings, so if this is a problem moving forward, I can augment it to read them in as floats.

The JSON reader and writer has also been completed, as well as the CSV reader and writer.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

The read_text and write_text functions are set up as well, and have been torture tested to some degree. One issue I noticed was that it did read in all the numbers as strings, so if this is a problem moving forward, I can augment it to read them in as floats.

The JSON reader and writer has also been completed, as well as the CSV reader and writer.

There is also a function that automatically formats the header in the excel sheet, so that the column names are 70 degrees tilted, as well as in bold text.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

The read_text and write_text functions are set up as well, and have been torture tested to some degree. One issue I noticed was that it did read in all the numbers as strings, so if this is a problem moving forward, I can augment it to read them in as floats.

The JSON reader and writer has also been completed, as well as the CSV reader and writer.

There is also a function that automatically formats the header in the excel sheet, so that the column names are 70 degrees tilted, as well as in bold text. The columns will also be resized in case of long entries.

There is also a sorting function that will represent the sample names in numerical order.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

The read_text and write_text functions are set up as well, and have been torture tested to some degree. One issue I noticed was that it did read in all the numbers as strings, so if this is a problem moving forward, I can augment it to read them in as floats.

The JSON reader and writer has also been completed, as well as the CSV reader and writer.

There is also a function that automatically formats the header in the excel sheet, so that the column names are 70 degrees tilted, as well as in bold text. The columns will also be resized in case of long entries.

There is also a sorting function that will represent the sample names in numerical order.

Commit:
Modified .gitignore it ignore venv and added requirements
to requirements.txt.

Commit:
Fixed some relative path imports (from excel import ... -->
from .excel import ...).

I also discovered a few more dependencies, so requirements.txt
was also updated.

Finally, confluence.io.base.BaseReader was not being used,
and was removed.

amikulichmines committed:
The goals of this code are:
1. Read in/Parse an arbitrary number of Excel files (workbooks).
2. Merge the contents from each identically named worksheet in those workbooks. (This is known more generally as “group by”. We must group by worksheet name.)
3. Write a single workbook with multiple worksheets. Each output worksheet contains the merged contribution from the worksheets from each workbook that share a common name. The worksheets in the final workbook will be the union of worksheets from all workbooks parsed in step #1.

The main() function is where the files are parsed in. In the write_excel function, data from identically named worksheets are grouped together and analyzed for merge conflicts.

All of the argparse functions are written, including '-i' to specify input file type, '-o' to specify the output file, '--interactive' to let the user chose the correct value in case of a merge conflict, '--mergedefault' to specify the default action in case a merge conflict arises, and '-q' to quietly abort the merge if there is a conflict.

The read_text and write_text functions are set up as well, and have been torture tested to some degree. One issue I noticed was that it did read in all the numbers as strings, so if this is a problem moving forward, I can augment it to read them in as floats.

The JSON reader and writer has also been completed, as well as the CSV reader and writer.

There is also a function that automatically formats the header in the excel sheet, so that the column names are 70 degrees tilted, as well as in bold text. The columns will also be resized in case of long entries.

There is also a sorting function that will represent the sample names in numerical order.

The test_merge functions are now fixed to include 87% of the functions. It still needs to be updated to test the argparse functions.

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

connorweddle2 committed:
Trying to get the merge.py to run on my computer by changing file path. Still receiving error:

Traceback (most recent call last):
  File "merge.py", line 24, in <module>
    from confluence.excel import ExcelReader
ModuleNotFoundError: No module named 'confluence'

connorweddle2 committed:
Revert "Trying to get the merge.py to run on my computer by changing file path. Still receiving error:"

This reverts commit 5e26a3682cd9641e57f1f254e4f2a1439d9729fc.

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

amikulichmines committed:
Merge remote-tracking branch 'origin/am.test_qm_metadata' into am.test_qm_metadata

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

connorweddle2 committed:
Changed Slashes

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

# Conflicts:
#	src/confluence/excel.py
#	src/confluence/merge.py

connorweddle2 committed:
CLI now works, along with all tests. Test added to check for correct output when different data types are used in input files.

connorweddle2 committed:
Added an option to take average of two values if a merge conflict occurs.

connorweddle2 committed:
Deleted test_excel_reader.

amikulichmines committed:
Added function to test files with multiple sheetnames

connorweddle2 committed:
Added CLI tests to test_merge using subprocess library

connorweddle2 committed:
Cleaned up file structure for test files.

connorweddle2 committed:
Changed merge function name to merge_files. Added CLI and Ingestor files.

bkappes commented on src/confluence/merge.py:
```diff
@@ -29,7 +29,7 @@ def run(args):
+def merge_files(args, sheetname='Sheet1'):
     """
     Function: This does the same thing as 'run()' but instead of writing the dataframes to a file, it returns the
     dataframe.
```
What are `args`? Please finish filling out the docstring.

bkappes commented on src/confluence/merge.py:
```diff
@@ -29,7 +29,7 @@ def run(args):
 
 
-def merge(args, sheetname='Sheet1'):
+def merge_files(args, sheetname='Sheet1'):
```
How does this handle workbooks with multiple sheetnames?

bkappes reviewed (commented):
I had a couple questions on the logic around `merge_files`.

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

connorweddle2 committed:
Added CLI(confluence.py) with merge option

connorweddle2 committed:
Added setup tools to run confluence CLI

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

amikulichmines committed:
Merge remote-tracking branch 'origin/am.test_qm_metadata' into am.test_qm_metadata

# Conflicts:
#	src/confluence/merge.py

connorweddle2 committed:
Added list_items command line option

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

connorweddle2 committed:
Changed CLI argparse structure

connorweddle2 committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

connorweddle2 committed:
Edited key parser argument

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

 Conflicts:
	src/confluence/confluence_CLI.py

connorweddle2 committed:
Added pif_maker

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

connorweddle2 committed:
Updated CLI implementation using subparsers

connorweddle2 committed:
Moved CSV.py, excel.py, JSON.py and text.py to io

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

 Conflicts:
	src/confluence/confluence_CLI.py
	src/confluence/merge.py
	src/confluence/pif_maker.py

connorweddle2 committed:
Changed subparser arguments for list_parser

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

 Conflicts:
	src/confluence/list_items.py

connorweddle2 committed:
Added action key argument to list_parser

amikulichmines committed:
Added function to see if there was no output file selected

Made the program able to parse empty files

Added empty file test to test_merge.py

Fixed the recursive function

Made one more fix, altered merge_files() to set the default variables

Fixed the check_for_sample_name_uniqueness file to have an abort, join, and mean option

Fixed the 'dropna' function that got rid of empty columns

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

 Conflicts:
	src/confluence/confluence_CLI.py

amikulichmines committed:
Fixed list duplicates function

connorweddle2 committed:
Updated CLI Parsing Method

amikulichmines committed:
Created pif classes

amikulichmines committed:
Created pif classes

connorweddle2 committed:
Updated CLI parsing method

connorweddle2 committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

connorweddle2 committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

amikulichmines committed:
Created pif classes
Fixed the tests

amikulichmines committed:
Created pif classes
Fixed the tests

amikulichmines committed:
Merge remote-tracking branch 'origin/am.test_qm_metadata' into am.test_qm_metadata

amikulichmines committed:
Created pif classes
Fixed the tests

amikulichmines committed:
Created pif classes
Fixed the tests

connorweddle2 committed:
Changed confluence_CLI.py to confluence.py

amikulichmines committed:
Merge branch 'am.test_qm_metadata' of https://github.com/csm-adapt/confluence into am.test_qm_metadata

 Conflicts:
	src/confluence/confluence.py

amikulichmines committed:
Created pif classes
Fixed the tests
